pkg/controller/utils/utils.go

```diff
@@ -174,6 +175,41 @@ func AddSecretWatchWithLabel(c controller.Controller, ns, label string) error {
 	})
 }
 
+// AddCSRWatchWithLabel adds a watch for CSRs with the given label.
+func AddCSRWatchWithLabel(c controller.Controller, labelName string) error {
+	return c.Watch(&source.Kind{Type: &certificatesv1.CertificateSigningRequest{}}, &handler.EnqueueRequestForObject{}, &predicate.Funcs{
+		CreateFunc: func(e event.CreateEvent) bool {
+			_, hasLabel := e.Object.GetLabels()[labelName]
+			return hasLabel
+		},
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			_, hasLabel := e.ObjectNew.GetLabels()[labelName]
+			if !hasLabel {
+				return false
+			}
+			csr, ok := e.ObjectNew.(*certificatesv1.CertificateSigningRequest)
+			if !ok {
+				return false
+			}
+			if csr.Spec.SignerName != OperatorCSRSignerName {
+				return false
+			}
+			for _, condition := range csr.Status.Conditions {
+				if (condition.Type == certificatesv1.CertificateDenied || condition.Type == certificatesv1.CertificateFailed) && condition.Status == corev1.ConditionTrue {
```

Comment:
It's a bit complicated that we have this check in the watch as well as in the CSR controller itself.

Can we commonize this logic into something like `func DoWeCareAboutThisCSR(obj) bool` that we can use in both places? That would make it much harder for these to get out of sync.